packages/editor/src/browser/editor-quick-open.ts

```diff
@@ -0,0 +1,151 @@
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            placeholder: '',
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            },
+            onClose: () => {
+                this.cancelIndicator.cancel();
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    public async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        this.cancelIndicator.cancel();
+        this.cancelIndicator = new CancellationTokenSource();
+        const token = this.cancelIndicator.token;
+
+        const editorItems: QuickOpenItem[] = [];
+
+        // Get URIs of all EditorWidgets that are currently opened, sorted by URI
+        const widgets: URI[] = this.editorManager.getAllTrackedEditors()
+            .filter((w: EditorWidget) => !!w.getResourceUri())
+            .map((w: EditorWidget) => w.getResourceUri()!)
+            .sort();
+
+        if (widgets.length === 0) {
+            editorItems.push(new QuickOpenItem({
+                label: 'List of opened editors is currently empty',
+                run: () => false
+            }));
+            acceptor(editorItems);
+            return;
+        }
+
+        for (const uri of widgets) {
+            if (uri.scheme === 'file') {
+                const item = await this.toItem(uri);
+                if (!!token.isCancellationRequested) {
+                    return;
+                }
+                editorItems.push(item);
+            }
+            acceptor(editorItems);
+        }
+        return;
+    }
+
+    private async toItem(uri: URI, group?: QuickOpenGroupItemOptions): Promise<QuickOpenItem<QuickOpenItemOptions>> {
+        const description = this.labelProvider.getLongName(uri.parent);
+
+        const options: QuickOpenItemOptions = {
+            label: this.labelProvider.getName(uri),
+            iconClass: await this.labelProvider.getIcon(uri) + ' file-icon',
+            description: description,
+            tooltip: uri.path.toString(),
+            uri: uri,
+            hidden: false,
+            run: this.getRunFunction(uri)
+        };
+        return new QuickOpenItem<QuickOpenItemOptions>(options);
+    }
+
+    protected getRunFunction(uri: URI): (mode: QuickOpenMode) => boolean {
+        return (mode: QuickOpenMode) => {
+            if (mode !== QuickOpenMode.OPEN) {
+                return false;
+            }
+            this.openFile(uri);
+            return true;
+        };
+    }
+
+    openFile(uri: URI): void {
+        this.openerService.getOpener(uri)
+            .then(opener => opener.open(uri))
+            .catch(error => this.messageService.error(error));
```

Comment:
please either log or don't catch, showing stacktraces to a user with notifications is not good

Reply:
Sure I will remove that.